### PR TITLE
Management security for TLS and certs added

### LINKED ISSUE #662: Add option for management security to disable TLS 1.0/1.1 and use certs
**Issue Type**

- [x] Template enhancement
- [x] Role enhancement
- [x] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

For complying with security hardening guide in EOS central https://eos.arista.com/arista-eos-hardening-guide/ we would need to disable TLS 1.0 and 1.1, and use only TLS 1.2.

For this we will need to create ssl profile and be able to enable/disable the above, additionally use certs and so on.

**Describe the solution you'd like**

```
management security
   ssl profile test
      tls versions 1.2
```
Also for certificate

```
(config-mgmt-sec-ssl-profile-test)#?
  certificate  Configure certificate for self authentication
  cipher-list  Configure a cipher list
  trust        Configure trusted certificate
```

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xaviramon**
## Change Summary
- Management security update to include tls version and certificates under a SSL Profile
- Management API HTTPS update to include SSL Profile
- Code modification for proper order in configuration and use of arista.avd.defined

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes issue #662 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Checklist:

- [ ] My code has been rebased from devel before I start
- [ ] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-api-http.j2 [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
 
 | VRF Name | IPv4 ACL | IPv6 ACL |
 | -------- | -------- | -------- |
 {%         for vrf in management_api_http.enable_vrfs | arista.avd.natural_sort %}
-| {{ vrf }} | {% if management_api_http.enable_vrfs[vrf].access_group is defined and management_api_http.enable_vrfs[vrf].access_group is not none %} {{ management_api_http.enable_vrfs[vrf].access_group }} {% else %} - {% endif %} | {% if management_api_http.enable_vrfs[vrf].ipv6_access_group is defined and management_api_http.enable_vrfs[vrf].ipv6_access_group is not none %} {{ management_api_http.enable_vrfs[vrf].ipv6_access_group }} {% else %} - {% endif %} |
+| {{ vrf }} | {% if management_api_http.enable_vrfs[vrf].access_group is arista.avd.defined %} {{ management_api_http.enable_vrfs[vrf].access_group }} {% else %} - {% endif %} | {% if management_api_http.enable_vrfs[vrf].ipv6_access_group is arista.avd.defined %} {{ management_api_http.enable_vrfs[vrf].ipv6_access_group }} {% else %} - {% endif %} |
 {%         endfor %}
 {%     endif %}
 
+{%  if management_api_http.enable_https is arista.avd.defined and management_api_http.enable_https == true and management_api_http.https_ssl_profile is arista.avd.defined %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-api-http.j2]: This block `management_api_http.enable_https is arista.avd.defined and management_api_http.enable_https == true` can be reduced to `management_api_http.enable_https is arista.avd.defined(true)`

**Review by titom73 — CHANGES_REQUESTED** [review]

**Comment by titom73** [comment]
CI MKDOCS issue is related to a link change and will be fixed in #698

**Review by titom73 — APPROVED** [review]
LGTM!

**Comment by ClausHolbechArista** [comment]
we should add
```yaml
protocol_https_certificate:
  certificate: < full certificate as multi-line PEM string >
  private_key: < full private key as multi-line PEM string >
```

Should be rendered as:
```eos
management api https-commands
   protocol https certificate
   {{ protocol_https_certificate.certificate }}
   EOF
   {{ protocol_https_certificate.private_key }}
   EOF
```

**Comment by carlbuchmann** [comment]
replaced by PR #780

**Pull request closed without merging**